prow/plugins/cat/cat_test.go

```diff
@@ -121,6 +121,15 @@ func TestUrl(t *testing.T) {
 			movie:    true,
 			require:  []string{"category=this", "api_key=that", "&", "mime_types=gif"},
 		},
+		{
+			name:     "grumpy cat",
+			url:      "http://foo",
+			category: "no",
+			key:      "that",
+			movie:    false,
+			require:  []string{"category=no"},
+			deny:     []string{"mime_types=gif"},
```

Comment:
Whoops, looks like TestURL uses `realClowder`, but it does not actually call the `readCat` method 🤦‍♂ Sorry about that!
I think it might be easiest to test the new functionality of `realClowder.readCat()` with a new `TestGrumpy` function specific to this use case.